client/java/src/main/java/com/linecorp/centraldogma/client/Watcher.java

```diff
@@ -58,9 +61,21 @@
     }
 
     /**
-     * Waits for the initial value to be available.
+     * Waits for the initial value to be available. Specify the default value with
+     * {@link #awaitInitialValue(long, TimeUnit, Object)} or make sure to handle
+     * a {@link TimeoutException} properly if the initial value must be available
+     * even when the server is unavailable.
+     *
+     * @param timeout the maximum amount of time to wait for the initial value. It is recommended to use a
+     *                value larger than 30 seconds at least so that the client can retry at least a few times
```

Comment:
Where does this `30s` value come from? CD server or client settings?
Asking because it sounds extremely big for a modern system.
What would you recommend our users if they need to shorten it?

Reply:
A user could specify a short timeout and a default value? The recommended timeout is to ensure that there are at least 4-5 retries in case of unavailability, because the client does exponential backoff all the way up from 2 seconds to 1 minute, i.e. t0 -> t2 -> t4 -> t8 -> t16. I'm not sure if we need to expose this level of detail, though.

Reply:
> the client does exponential backoff all the way up from 2 seconds to 1 minute

Thanks for the info. Just FMI, is this backoff (initial and factor) configurable?

Reply:
Updated the Javadoc and client library manual to reflect your feed back. Please check again. :bowing_man: 

> Just FMI, is this backoff (initial and factor) configurable?

The backoff is currently hard-coded. Do you think there's a case a user would want to use non-default backoff? I'm somewhat concerned about cargo-culting too short retry interval.

Reply:
> Do you think there's a case a user would want to use non-default backoff?

CD is used for many purposes and I don't dare say I know them all :).

I could see a use case for disabling the retries (failing immediately and letting end-user manage the retry (or not) at his own convenience.
In our flagship server project, we have for instance
- sampling configurations : totally fine with immediate failure/fallback default value, we can peek again when CD goes back online
- server repertory : can't boot without it
If the end-user's use case is all of the first kind, then the time spent sleeping between retries is a waste. WDYT?

Reply:
> sampling configurations

In that case, you don't really need to use `awaitInitialValue()` or you can use a very short timeout, and then peek the latest value using `latestValue(defaultValue)` which never blocks. If the server is unavailable at startup, it will start with the default value, but once the server is back up, `latestValue()` will start to return the latest value retrieved from the server.

Reply:
Let me add some note about this case as well in `client-java.rst`. Thanks for a good idea.

Reply:
Thanks! It sounds like our usage of `awaitInitialValue` in both use cases (mostly from copy-pasting, I dare say) was a bit wrong. /cc @imasahiro

Reply:
Please check the updated docs :-)

Reply:
Thanks, LGTM